### PR TITLE
Fix body vs symmetry subgraph filtering

### PR DESCRIPTION
**Pull request by talmo**
### Description
Currently, we may not be able to train bottom-up models from the GUI if the skeleton contains symmetry edges that would result in it not being an arborescence.

This is fixed by using only the skeleton subgraph that contains body (non-symmetry) edges to test for whether it is an arborescence.

This PR uses the appropriate subgraph for arborescence testing, as well as fixing how we induced the subgraph views for body vs symmetry edges. Previously, we also had a subtle edge case where calling `Skeleton.graph` would return an empty graph if there were nodes but no body edges. We now return disconnected nodes appropriately.

### Types of changes

- [x] Bugfix
- [ ] New feature
- [ ] Refactor / Code style update (no logical changes)
- [ ] Build / CI changes
- [ ] Documentation Update
- [ ] Other (explain)

### Does this address any currently open issues?
[list open issues here]

### Outside contributors checklist

- [ ] Review the [guidelines for contributing](https://github.com/talmolab/sleap/blob/develop/docs/CONTRIBUTING.md) to this repository
- [ ] Read and sign the [CLA](https://github.com/talmolab/sleap/blob/develop/sleap-cla.pdf) and add yourself to the [authors list](https://github.com/talmolab/sleap/blob/develop/AUTHORS)
- [ ] Make sure you are making a pull request against the **develop** branch (not *main*). Also you should start *your branch* off *develop*
- [ ] Add tests that prove your fix is effective or that your feature works
- [ ] Add necessary documentation (if appropriate)

#### Thank you for contributing to SLEAP!
:heart:

### CODE REVIEW

File: sleap/skeleton.py [hunk 1]
```diff
@@ -171,42 +171,39 @@ def dict_match(dict1, dict2):
...
-        ]
-        # TODO: properly induce subgraph for MultiDiGraph
-        #   Currently, NetworkX will just return the nodes in the subgraph.
-        #   See: https://stackoverflow.com/questions/16150557/networkxcreating-a-subgraph-induced-from-edges
-        return self._graph.edge_subgraph(edges)
+        """Return a view on the subgraph of body nodes and edges for skeleton."""
+
+        def edge_filter_fn(src, dst, edge_key):
+            edge_data = self._graph.get_edge_data(src, dst, edge_key)
+            return edge_data["type"] == EdgeType.BODY
```
Comment (roomrys) [inline sleap/skeleton.py:194]: Did you want to make this into a method so we can re-use it?

File: sleap/skeleton.py [hunk 2]
```diff
@@ -171,42 +171,39 @@ def dict_match(dict1, dict2):
...
-        edges = [
-            (src, dst, key)
-            for src, dst, key, edge_type in self._graph.edges(keys=True, data="type")
-            if edge_type == EdgeType.SYMMETRY
-        ]
-        return self._graph.edge_subgraph(edges)
+
+        def edge_filter_fn(src, dst, edge_key):
+            edge_data = self._graph.get_edge_data(src, dst, edge_key)
+            return edge_data["type"] == EdgeType.SYMMETRY
```
Comment (roomrys) [inline sleap/skeleton.py:204]: Ditto above: Did you want to make this into a method so we can re-use it?

**Review by roomrys — APPROVED** [review]
Looks good, small question

**Comment by talmo** [comment]
> Looks good, small question

Honestly it's such a small snippet of mostly networkx code and we exclusively use it for these two cases, so probably not really worth it. We only have two edge types in the skeleton and I can't foresee us having to add more, so I think this is ok since they're both pretty well tested.

**Pull request merged**